File: Prak2-Inheritance/Expression/AddExpression.cpp
/* Nama         : Gregorius Jovan Kresnadi */
/* NIM          : 13518135 */
/* Tanggal      : 13-02-2020 */

#include "AddExpression.hpp"
#include <iostream>

using namespace std;

AddExpression::AddExpression(Expression* x, Expression* y) : BinaryExpression(x,y) {
}

int AddExpression::solve() {
    return x->solve() + y->solve();
}

File: PrakJ1-Intro/T/2/AddExpression.java
// Nama         : Gregorius Jovan Kresnadi
// NIM          : 13518135
// Tanggal      : 12-03-2020
// File 		: AddExpression.java

class AddExpression implements Expression {
	protected Expression x;
	protected Expression y;

	public AddExpression(Expression x, Expression y) {
		this.x = x;
		this.y = y;
	}

	public int solve() {
		return x.solve() + y.solve();
	}
}